File: python/problem025.py
fibonacci = [0, 1, 1]


def set_fibonacci(n):
    fibonacci.append(fibonacci[n - 1] + fibonacci[n - 2])

result = 2
while len(str(fibonacci[result])) < 1000:
    result += 1
    set_fibonacci(result)

print(result)

File: java/src/Problem025.java
import java.math.BigInteger;

public class Problem025 {

    BigInteger fibonacciOne = new BigInteger("1");
    BigInteger fibonacciTwo = new BigInteger("2");

    int result(){
        int index = 3;
        while(fibonacciTwo.toString().length() < 1000){
            BigInteger fibonacciTemp = fibonacciTwo;
            fibonacciTwo = fibonacciOne.add(fibonacciTwo);
            fibonacciOne = fibonacciTemp;
            index = index + 1;
        }
        return index;
    }

    public static void main(String args[]){
        Problem025 problem = new Problem025();
        System.out.println(problem.result());
    }
}
